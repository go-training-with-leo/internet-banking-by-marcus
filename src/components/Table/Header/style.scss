@import "assets/scss/mixins.scss";
@import "assets/scss/variables.scss";

.table-header {
  border-bottom: #000 3px solid;
}
