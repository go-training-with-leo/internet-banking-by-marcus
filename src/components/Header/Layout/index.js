import React from 'react';
import PropTypes from 'prop-types';

import { NotifyBusy, NotifyRemove, NotifyFree } from 'assets/images';

import './style.scss';

const Header = ({ notifyBusy, notifyFree, notifyRemove, title, children }) => {
  return (
    <div className='header'>
      <div className='header-left'>
        <span>{title}</span>
        {children}
      </div>
      {notifyBusy ? (
        <NotifyBusy className='notif-busy' width={40} height={40} />
      ) : notifyRemove ? (
        <NotifyRemove className='notif-remove' width={40} height={40} />
      ) : (
        notifyFree && (
          <NotifyFree className='notif-free' width={40} height={40} />
        )
      )}
    </div>
  );
};

Header.defaultProps = {
  notifyBusy: false,
  notifyFree: false,
  notifyRemove: false,
  title: undefined,
  children: undefined,
};

Header.propsTypes = {
  notifyBusy: PropTypes.bool,
  notifyFree: PropTypes.bool,
  notifyRemove: PropTypes.bool,
  title: PropTypes.string,
  children: PropTypes.node,
};

export default Header;
