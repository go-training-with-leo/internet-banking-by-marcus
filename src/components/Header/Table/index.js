import React from 'react';
import PropTypes from 'prop-types';

import './style.scss';

const HeaderTable = ({ children }) => {
  return (
    <tr className='table-header'>
      <th>&nbsp;</th>
      {children}
    </tr>
  );
};

HeaderTable.defaultProps = {
  children: undefined,
};

HeaderTable.propTypes = {
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.array]),
};

export default HeaderTable;
