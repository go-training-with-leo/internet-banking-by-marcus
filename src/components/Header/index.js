import React from 'react';

import 'components/Header/style.scss';

const Header = () => {
  return (
    <div className='header'>
      <div>Header</div>
    </div>
  );
};

export default Header;
