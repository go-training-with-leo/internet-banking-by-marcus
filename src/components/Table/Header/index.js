import PropTypes from 'prop-types';
import React from 'react';

import './style.scss';

const HeaderTable = ({ children }) => {
  return (
    <tr className='table-header'>
      <th>&nbsp;</th>
      {children}
    </tr>
  );
};

HeaderTable.defaultProps = {
  children: undefined,
};

HeaderTable.propTypes = {
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.array]),
};

export default HeaderTable;
