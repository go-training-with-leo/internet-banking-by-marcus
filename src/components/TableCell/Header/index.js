import React from 'react';
import PropTypes from 'prop-types';

import './style.scss';

const HeaderCell = ({ children, key }) => {
  return (
    <th className='table-header-cell' key={key}>
      <div>{children}</div>
    </th>
  );
};

HeaderCell.defaultProps = {
  children: undefined,
};

HeaderCell.propTypes = {
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.array]),
};

export default HeaderCell;
