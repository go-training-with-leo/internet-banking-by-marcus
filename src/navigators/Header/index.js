import PropTypes from 'prop-types';
import React from 'react';

import { NotifyBusy, NotifyRemove, NotifyFree, Menu } from 'assets/images';

import Notification from 'components/Notification';
import useToggle from 'components/hooks/useToggle';

import './style.scss';

const messages = [
  {
    id: 'MSG1',
    account: {
      type: 'lender',
      name: 'Jusin Doe',
      last4Digit: '4968',
    },
    type: 'removed',
    time: '09:51:36 23/05/2020',
  },
  {
    id: 'MSG2',
    account: {
      type: 'debtor',
      name: 'Jusin Doe',
      last4Digit: '4968',
    },
    type: 'repaid',
    time: '09:51:36 23/05/2020',
  },
  {
    id: 'MSG3',
    account: {
      type: 'lender',
      name: 'Jusin Doe',
      last4Digit: '4968',
    },
    type: 'removed',
    time: '09:51:36 23/05/2020',
  },
  {
    id: 'MSG4',
    account: {
      type: 'lender',
      name: 'Jusin Doe',
      last4Digit: '4968',
    },
    type: 'removed',
    time: '09:51:36 23/05/2020',
  },
  {
    id: 'MSG5',
    account: {
      type: 'lender',
      name: 'Jusin Doe',
      last4Digit: '4968',
    },
    type: 'removed',
    time: '09:51:36 23/05/2020',
  },
  {
    id: 'MSG6',
    account: {
      type: 'lender',
      name: 'Jusin Doe',
      last4Digit: '4968',
    },
    type: 'removed',
    time: '09:51:36 23/05/2020',
  },
  {
    id: 'MSG7',
    account: {
      type: 'lender',
      name: 'Jusin Doe',
      last4Digit: '4968',
    },
    type: 'removed',
    time: '09:51:36 23/05/2020',
  },
];

const Header = ({ type, title, children, onMenuClick }) => {
  const [showNotif, setShowNotif] = useToggle();

  const notifycation = {
    busy: (
      <NotifyBusy
        className='notif-busy'
        width={40}
        height={40}
        onClick={setShowNotif}
      />
    ),
    remove: (
      <NotifyRemove
        className='notif-remove'
        width={40}
        height={40}
        onClick={setShowNotif}
      />
    ),
    free: (
      <NotifyFree
        className='notif-free'
        width={40}
        height={40}
        onClick={setShowNotif}
      />
    ),
  };
  return (
    <div className='header'>
      <div className='header-left'>
        <Menu className='mobile-menu' fill='white' onClick={onMenuClick} />
        <span>{title}</span>
        {children}
      </div>
      <div className='notif hide'>
        {notifycation[type]}
        {showNotif && <Notification messages={messages} />}
      </div>
    </div>
  );
};

Header.defaultProps = {
  type: '',
  title: undefined,
  children: undefined,
};

Header.propsTypes = {
  type: PropTypes.oneOf(['free', 'busy', 'remove']),
  title: PropTypes.string,
  children: PropTypes.node,
};

export default Header;
