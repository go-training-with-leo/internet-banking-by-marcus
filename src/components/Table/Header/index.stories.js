import React from 'react';

import TableCell from 'components/Table/HeaderCell';
import { PlusIcon } from 'assets/images';
import Header from '.';

export default {
  title: 'Page Header',
  component: Header,
};

const HeaderTable = () => (
  <Header>
    <TableCell>
      <span>Header</span>
      <PlusIcon />
    </TableCell>
  </Header>
);

export { HeaderTable };
