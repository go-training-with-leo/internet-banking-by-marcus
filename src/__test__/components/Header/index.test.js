import React from 'react';
import { render } from '@testing-library/react';

import Header from 'navigators/Header';

test('Test props', () => {
  const tree = render(
    <Header title='Title' type='busy'>
      <button>Button</button>
    </Header>
  );

  const header = tree.container.querySelector('div.header');

  expect(header).toHaveTextContent('Button');
  expect(header).toHaveTextContent('Title');
  expect(header.childElementCount).toEqual(2);
  expect(tree).toMatchSnapshot();
});

test('Test notifyBusy props order', () => {
  const tree = render(
    <Header title='Title' type='busy'>
      <button>Button</button>
    </Header>
  );

  const header = tree.container.querySelector('div.header');

  expect(header.lastElementChild).toHaveClass('notif-busy');
  expect(header.childElementCount).toEqual(2);
  expect(tree).toMatchSnapshot();
});

test('Test notifyRemove props order', () => {
  const tree = render(
    <Header title='Title' type='remove'>
      <button>Button</button>
    </Header>
  );

  const header = tree.container.querySelector('div.header');

  expect(header.lastElementChild).toHaveClass('notif-remove');
  expect(header.childElementCount).toEqual(2);
  expect(tree).toMatchSnapshot();
});
test('Test notifyFree props order', () => {
  const tree = render(
    <Header title='Title' type='free'>
      <button>Button</button>
    </Header>
  );

  const header = tree.container.querySelector('div.header');

  expect(header.lastElementChild).toHaveClass('notif-free');
  expect(header.childElementCount).toEqual(2);
  expect(tree).toMatchSnapshot();
});
