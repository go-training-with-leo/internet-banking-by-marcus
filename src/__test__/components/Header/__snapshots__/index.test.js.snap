// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test notifyBusy props order 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header"
      >
        <div
          class="header-left"
        >
          <svg
            class="mobile-menu"
            fill="white"
          >
            menu.svg
          </svg>
          <span>
            Title
          </span>
          <button>
            Button
          </button>
        </div>
        <div
          class="notif hide"
        >
          <svg
            class="notif-busy"
            height="40"
            width="40"
          >
            notificationBusy.svg
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      <div
        class="header-left"
      >
        <svg
          class="mobile-menu"
          fill="white"
        >
          menu.svg
        </svg>
        <span>
          Title
        </span>
        <button>
          Button
        </button>
      </div>
      <div
        class="notif hide"
      >
        <svg
          class="notif-busy"
          height="40"
          width="40"
        >
          notificationBusy.svg
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test notifyFree props order 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header"
      >
        <div
          class="header-left"
        >
          <svg
            class="mobile-menu"
            fill="white"
          >
            menu.svg
          </svg>
          <span>
            Title
          </span>
          <button>
            Button
          </button>
        </div>
        <div
          class="notif hide"
        >
          <svg
            class="notif-free"
            height="40"
            width="40"
          >
            notificationFree.svg
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      <div
        class="header-left"
      >
        <svg
          class="mobile-menu"
          fill="white"
        >
          menu.svg
        </svg>
        <span>
          Title
        </span>
        <button>
          Button
        </button>
      </div>
      <div
        class="notif hide"
      >
        <svg
          class="notif-free"
          height="40"
          width="40"
        >
          notificationFree.svg
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test notifyRemove props order 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header"
      >
        <div
          class="header-left"
        >
          <svg
            class="mobile-menu"
            fill="white"
          >
            menu.svg
          </svg>
          <span>
            Title
          </span>
          <button>
            Button
          </button>
        </div>
        <div
          class="notif hide"
        >
          <svg
            class="notif-remove"
            height="40"
            width="40"
          >
            notificationRemove.svg
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      <div
        class="header-left"
      >
        <svg
          class="mobile-menu"
          fill="white"
        >
          menu.svg
        </svg>
        <span>
          Title
        </span>
        <button>
          Button
        </button>
      </div>
      <div
        class="notif hide"
      >
        <svg
          class="notif-remove"
          height="40"
          width="40"
        >
          notificationRemove.svg
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header"
      >
        <div
          class="header-left"
        >
          <svg
            class="mobile-menu"
            fill="white"
          >
            menu.svg
          </svg>
          <span>
            Title
          </span>
          <button>
            Button
          </button>
        </div>
        <div
          class="notif hide"
        >
          <svg
            class="notif-busy"
            height="40"
            width="40"
          >
            notificationBusy.svg
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      <div
        class="header-left"
      >
        <svg
          class="mobile-menu"
          fill="white"
        >
          menu.svg
        </svg>
        <span>
          Title
        </span>
        <button>
          Button
        </button>
      </div>
      <div
        class="notif hide"
      >
        <svg
          class="notif-busy"
          height="40"
          width="40"
        >
          notificationBusy.svg
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
